fuel-relayer/src/mock_db.rs

```diff
@@ -14,68 +13,40 @@ use std::{
 
 use fuel_core_interfaces::{
     common::{
-        fuel_storage::{
+        fuel_tx::MessageId,
+        prelude::{
             StorageInspect,
             StorageMutate,
         },
-        fuel_tx::{
-            Address,
-            MessageId,
-        },
     },
     db::{
-        DelegatesIndexes,
         KvStoreError,
         Messages,
-        StakingDiffs,
-        ValidatorsSet,
     },
     model::{
         BlockHeight,
-        ConsensusId,
         DaBlockHeight,
         Message,
         SealedFuelBlock,
-        ValidatorId,
-        ValidatorStake,
-    },
-    relayer::{
-        RelayerDb,
-        StakingDiff,
-        ValidatorSet,
     },
+    relayer::RelayerDb,
 };
 
 #[derive(Default)]
-pub(crate) struct Data {
+pub struct Data {
     pub messages: HashMap<MessageId, Message>,
-    pub validators: HashMap<ValidatorId, (ValidatorStake, Option<ConsensusId>)>,
-    pub delegator_index: HashMap<Address, Vec<DaBlockHeight>>,
-    pub staking_diffs: BTreeMap<DaBlockHeight, StakingDiff>,
-
     pub chain_height: BlockHeight,
     pub sealed_blocks: HashMap<BlockHeight, Arc<SealedFuelBlock>>,
-    pub validators_height: DaBlockHeight,
-    pub finalized_da_height: DaBlockHeight,
+    pub finalized_da_height: Option<DaBlockHeight>,
     pub last_committed_finalized_fuel_height: BlockHeight,
+    pub pending_committed_fuel_height: Option<BlockHeight>,
 }
 
-#[derive(Default)]
-pub(crate) struct MockDb {
+#[derive(Default, Clone)]
+pub struct MockDb {
```

Comment:
Now that this is exported outside the crate, it should have a doc comment describing its usage. Should it be hidden before the `test-helpers` feature? Are breaking changes to MockDb considered semver-breaking? If no, that must be documented as well.

Reply:
The entire module is behind either the `test or test-helpers` feature flag so it's definitely not semvar breaking.